lib/SetAPI/assembly/AssemblySetInterfaceV1.py

```diff
@@ -74,24 +74,26 @@ def _validate_save_set_params(
         :param params: parameters to the save_set function
         :type params: dict[str, Any]
         """
-        # TODO: add checks that only one copy of each assembly is in the set
-
         if params.get("data", None) is None:
-            err_msg = data_required(self.set_items_type())
+            err_msg = param_required("data")
```

Comment:
instead of having `data_required` and `ref_required` errors, just have a generic `param_required(param)` function